firefox-ios/Client/Frontend/Home/Homepage Rebuild/NewHomepageCompositionalLayout.swift

```diff
@@ -0,0 +1,74 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import Common
+
+/// Holds section layout logic for the new homepage as part of the rebuild project
+class NewHomepageSectionLayoutProvider {
+    private var logger: Logger
+
+    init(logger: Logger = DefaultLogger.shared) {
+        self.logger = logger
+    }
+
+    func createCompositionalLayout() -> UICollectionViewCompositionalLayout {
+        return UICollectionViewCompositionalLayout { (sectionIndex, environment) -> NSCollectionLayoutSection? in
```

Comment:
breaking this up in further with `Control + M (or Editor > Structure > Format to Multiple Lines)` leads to warnings, so kept this length

Reply:
this is under 120 chars, no? Switlint shouldn't be yelling anyway, so it's fine as is. I'm the only one who uses 90 as my "too long" zone, but our team's convention is 120 chars is fine.

Reply:
yes, swiftlint wasn't complaining, but still seems a bit long for me too. 🤣 wasn't sure if there was a better way